File: 0136-single-number/0136-single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        //Optimal solution
        //TC: O(n) - SC: O(1)
        int xor1 = 0, n = nums.size();
        for(int i = 0;i<n;i++){
            xor1 ^= nums[i];
        }
        return xor1;
        
    }
};

File: 0136-single-number/0136-single-number.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        Array.Sort(nums);
        int ans = 0;
        bool isUnique = true;
        for(int i = 0;i<nums.Length;i++){
            ans ^= nums[i];
        }
        return ans;
    }
}